### PR TITLE
Add accredited_projects and accredited_with_clinical_report_projects functions

### PR DESCRIPTION
**Pull request by callunity**
JIRA Ticket: GP-3731

- [x] Updates Changelog

`N/A` Updates developer documentation

This PR is in service of being able to write an alert that only considers Accredited with Clinical Report projects.

### CODE REVIEW

**Review by mlaszloffy — APPROVED** [review]

File: plugin-pinery/src/main/java/ca/on/oicr/gsi/shesmu/pinery/PinerySource.java [hunk 1]
```diff
@@ -744,6 +744,32 @@ public Set<String> clinicalProjects() {
...
+        .get()
+        .filter(
+            project -> project.getPipeline() != null && "Accredited".equals(project.getPipeline()))
+        .map(SampleProjectDto::getName)
+        .collect(Collectors.toCollection(TreeSet::new));
+  }
+
+  @ShesmuMethod(
+      name = "accredited_with_clinical_report_projects",
+      description = "Projects with Accredited pipeline from in Pinery defined in {file}.")
```
Comment (avarsava) [inline plugin-pinery/src/main/java/ca/on/oicr/gsi/shesmu/pinery/PinerySource.java:761]: Description could reflect 'with Report'

**Review by avarsava — COMMENTED** [review]
The 'from in Pinery' construction reads weird but it's consistent.
isn't 'Accredited with Report' replacing 'Clinical'?

**Comment by callunity** [comment]
> The 'from in Pinery' construction reads weird but it's consistent. isn't 'Accredited with Report' replacing 'Clinical'?

Ah yep, I'll update all those descriptions. Definitely copypasta'd that.
"Accredited with Clinical Report" and "Accredited" are both clinical pipelines, but only AwCR samples are "true" clinical samples.

File: plugin-pinery/src/main/java/ca/on/oicr/gsi/shesmu/pinery/PinerySource.java (hunk 1 above)
Comment (callunity) [reply plugin-pinery/src/main/java/ca/on/oicr/gsi/shesmu/pinery/PinerySource.java]: done

**Comment by avarsava** [comment]
> > The 'from in Pinery' construction reads weird but it's consistent. isn't 'Accredited with Report' replacing 'Clinical'?
> 
> Ah yep, I'll update all those descriptions. Definitely copypasta'd that. "Accredited with Clinical Report" and "Accredited" are both clinical pipelines, but only AwCR samples are "true" clinical samples.

Right, so just want to double check, does the `clinical_projects` method still make sense?

**Review by avarsava — APPROVED** [review]

**Pull request merged**